app/code/Magento/InventoryIndexer/Indexer/Stock/IndexDataProviderByStockId.php

```diff
@@ -22,20 +23,27 @@ class IndexDataProviderByStockId
     private $resourceConnection;
 
     /**
-     * @var SelectBuilder
+     * @var SelectManagableBuilder
      */
-    private $selectBuilder;
+    private $selectManagableBuilder;
+
+    /**
+     * @var SelectNotManagableBuilder
+     */
+    private $selectNotManagableBuilder;
 
     /**
      * @param ResourceConnection $resourceConnection
      * @param SelectBuilder $selectBuilder
      */
     public function __construct(
         ResourceConnection $resourceConnection,
-        SelectBuilder $selectBuilder
+        SelectBuilder $selectManagableBuilder,
+        SelectNotManagableBuilder $selectNotManagableBuilder
     ) {
         $this->resourceConnection = $resourceConnection;
-        $this->selectBuilder = $selectBuilder;
+        $this->selectManagableBuilder = $selectManagableBuilder;
+        $this->selectNotManagableBuilder = $selectNotManagableBuilder;
```

Comment:
Please handle this changes in backward compatible way. (pretty same situation as inpp/code/Magento/InventoryIndexer/Indexer/SourceItem/IndexDataBySkuListProvider.php)